main/tests.py

```diff
@@ -87,3 +88,44 @@ def test_unlogged_user_access_to_homepage(self, client):
         response = client.get("/homepage/")
         assert response.status_code == 302
         assert response['Location'] == '/login/?next=/homepage/'
+
+
+@pytest.mark.django_db
+class TestDogOwnerHomePageView:
+    def test_dog_owner_present_all_day_cares_at_entrypoint(self, client, create_dog_owner_user):
+        client.force_login(user=create_dog_owner_user.user)
+        response = client.get("/homepage/")
+        day_care_queryset = response.context['day_care_queryset']
+        assert day_care_queryset.count() == DayCare.objects.all().count()
```

Comment:
maybe instead of count use set(day_care_queryset) == set(daycare.objects.all) because if the count is the same we are not sure we presenting all daycares, maybe we present the same one 30 times?

Reply:
Im not sure I fully understand , but in this test we want to check that day_care_queryset contain all Daycare objects when we are entering to dog_owner homepage.
Im not sure that "set" is define on queryset and since every user is unique it is promise that every object in the queryset is diffrent from the other.

Reply:
that's right for the right side of the assert (daycare.objects), what about what is presented in HTML (daycare_queryset), how can you be sure that they are unique and don't have dups, for example, daycare shown twice in HTML or things like that

Reply:
And anyway it just seems weird to conclude equality between sets by their size, even if all the values are unique

Reply:
The HTML running for loop and create a card for each daycare in day_care query set so thats way I think is enough to test **what this queryset contatin** .
Since day_care_query is **subset** of all day_cares queryset, that means that if the number of elements between them is equal they are have to contatin the same values .

Reply:
But if u think it is less readble I will try your segestion about "set(day_care_queryset) == set(daycare.objects.all)" (: